OpenCL_C.txt

```diff
@@ -3180,6 +3180,20 @@ following forms whose meanings are described elsewhere:
 #pragma OPENCL EXTENSION all : behavior
 ----------
 
+`FP_CONTRACT` directive can be used to specify contraction for some floating
+point operations (more details can be found in <<floating-point-macros-and-pragmas,
+the floating-point macros and pragmas section>>).
+
+`EXTENSION` directive is used to enable logic in the compiler that deviates
+from the standard behavior described in this document. Extensions are not
+guaranteed to be supported by all implementations and their presence can be
+checked by availability of specific predefined macros. More information can be
+found in <<opencl-extension-spec,section 1.2 in the OpenCL Extension
+Specification>>. Pragma directives are only needed for extensions that require
+changes in the language such that the source code is interpreted differently
+with pragma and without. Each extensions should specify whether it adds support
+for pragma and what behavior will be changed due to its use in the source.
+
```

Comment:
Could this text be further clarified with the cases where pragma directives have no effect? A vendor may have platform/runtime extensions, and I assume the intent is that pragma directives cannot and should not be used to enable or disable those.

As a contrived example, I assume `pragma OPENCL EXTENSION cl_khr_icd : disable` is nonsensical, but it seems that the spec does actually allow for someone to put that in an OpenCL-C kernel.

Reply:
> Could this text be further clarified with the cases where pragma directives have no effect? A vendor may have platform/runtime extensions, and I assume the intent is that pragma directives cannot and should not be used to enable or disable those.

I would say that there shouldn't be any pragma with no effect at all. Becasue it is simply redundant. And therefore the current clarification says that the spec should explain what the pragma chages/modifies from the core specification. 

We could of course add something like that the pragma that doesn't provide any clarification is not guaranteed to be accepted. The important is that I would prefer to avoid saying that the pragma can alway be accepted but with no effect because it mean it has to be implemented in tooling  without any benefit for the developers.

> As a contrived example, I assume pragma OPENCL EXTENSION cl_khr_icd : disable is nonsensical, but it seems that the spec does actually allow for someone to put that in an OpenCL-C kernel.

FYI I looked into this in more detail and the only conclusion I have made is that sadly it is just a result of copy-paste. I have provided more details in https://github.com/KhronosGroup/OpenCL-Docs/issues/82#issuecomment-724703376 if you are interested. I hope that the pragma from those extensions can simply be removed.

Reply:
I have added the following:
> If the behavior of an extension pragma is unspecified the implementations are not
required to accept it. Moreover, there is no default behavior for the extension
pragma e.g. it does not define or undefine an associated predefined extension
macro.

Does it seem reasonable?

Reply:
Btw this clarification has been moved into ext/introduction.asciidoc

Reply:
It's a lot clearer now. Thanks!